src/EnergyPlus/HVACVariableRefrigerantFlow.cc

```diff
@@ -283,6 +283,29 @@ void SimulateVRF(EnergyPlusData &state,
         if (state.dataHVACVarRefFlow->VRF(VRFCondenser).CondenserType == DataHeatBalance::RefrigCondenserType::Water)
             UpdateVRFCondenser(state, VRFCondenser);
     }
+    if (state.dataHVACVarRefFlow->VRF(VRFCondenser).VRFAlgorithmType == AlgorithmType::FluidTCtrl) {
+        // consider cycling in fan calculation
+        for (int TUListNum = 1; TUListNum <= state.dataHVACVarRefFlow->NumVRFTULists; ++TUListNum) {
+            auto &thisTUList = state.dataHVACVarRefFlow->TerminalUnitList(TUListNum);
+            for (int TUNum = 1; TUNum <= thisTUList.NumTUInList; ++TUNum) {
+                int heatingCoilNum = state.dataHVACVarRefFlow->VRFTU(TUNum).HeatCoilIndex;
+                int coolingCoilNum = state.dataHVACVarRefFlow->VRFTU(TUNum).CoolCoilIndex;
+                auto &heatingCoil = state.dataDXCoils->DXCoil(heatingCoilNum);
+                auto &coolingCoil = state.dataDXCoils->DXCoil(coolingCoilNum);
+                int fanIndex = state.dataHVACVarRefFlow->VRFTU(TUNum).FanIndex;
+                // only deal with cooling for now. heating RTF might have some issue
+                // does Fan:SystemModel need adjustment as well? if so consider 0-indexing and it's in state.dataHVACFan->fanObjs vector
+                if (fanIndex > 0 && heatingCoil.HeatingCoilRuntimeFraction == 0.0) { // in cooling mode
+                    // here coolingCoil.CoolingCoilRuntimeFraction equals state.dataHVACVarRefFlow->VRF(VRFCond).VRFCondCyclingRatio
+                    // this is not the case for heating
+                    auto &fan = state.dataFans->Fan(fanIndex);
+                    fan.FanPower *= coolingCoil.CoolingCoilRuntimeFraction;
+                    fan.FanEnergy = fan.FanPower * state.dataHVACGlobal->TimeStepSysSec;
+                    fan.PowerLossToAir *= coolingCoil.CoolingCoilRuntimeFraction;
```

Comment:
I'm not sure I like this since here again one model is changing the result of another. The only fan models that change the "fans" result based on cycling is Fan:OnOff and Fan:SystemModel. Should this apply to a fan that operates the entire time step? The fan does not necessarily cycle with the compressor. The way this is done in other models is to set the state.dataHVACGlobal->OnOffFanPartLoadFraction variable and let the fan model decide what to do with power/energy.

    ZoneHVAC:TerminalUnit:VariableRefrigerantFlow,
      A7 ,  \field Supply Air Fan Object Type
            \type choice
            \key Fan:SystemModel
            \key Fan:OnOff
            \key Fan:ConstantVolume
            \key Fan:VariableVolume

Reply:
The VRF Fluid control model terminal unit only accepts a Fan:VariableVolume object (otherwise there will be an error message
```
Fan Object Type must be Fan:VariableVolume if VRF AirConditioner:VariableRefrigerantFlow:FluidTemperatureControl 
```
The `Fan:VariableVolume` object is simulated in the `SimVariableVolumeFan` function. But in this function, the `state.dataHVACGlobal->OnOffFanPartLoadFraction` was not used anywhere. So I modified the function to access this global variable which holds the coil runtime fraction.

Reply:
I agree with the idea of not applying a fan that operates for the entire time step and instead letting the fan model decide how to manage the power. Even though `state.dataHVACGlobal->OnOffFanPartLoadFraction` isn't used anywhere for now, it could be used elsewhere in the future . Any insights regarding this issue would be appreciated. @rraustad @mjwitte @EnergyArchmage @Myoldmopar

Reply:
I would agree. The cooling coil should set `state.dataHVACGlobal->OnOffFanPartLoadFraction`. And this should probably be done in a separate branch?

Reply:
I just checked CalcVRFCoolingCoil and CalcVRFCoolingCoil_FluidTCtrl and both already have this:

        // If cycling fan, send coil part-load fraction to on/off fan via HVACDataGlobals
        if (fanOp == HVAC::FanOp::Cycling) state.dataHVACGlobal->OnOffFanPartLoadFraction = PLF;

So there is nothing to do since the fan models already include this variable where necessary. If other fan types are used for VRF the code is already set up to model correctly.

    ZoneHVAC:TerminalUnit:VariableRefrigerantFlow,
      A7 ,  \field Supply Air Fan Object Type
        \type choice
        \key Fan:SystemModel
        \key Fan:OnOff
        \key Fan:ConstantVolume
        \key Fan:VariableVolume
        \default Fan:ConstantVolume

Reply:
The following is the note in field A7, it indicates that the `Fan:VariableVolume` fan is only allowed in `AirConditioner:VariableRefrigerantFlow:FluidTemperatureControl` and  `AirConditioner:VariableRefrigerantFlow:FluidTemperatureControl:HR`, not in the curve-based VRF terminal unit (`ZoneHVAC:TerminalUnit:VariableRefrigerantFlow`). 

```
        \note Supply Air Fan Object Type must be Fan:SystemModel, Fan:OnOff, or Fan:ConstantVolume
        \note if AirConditioner:VariableRefrigerantFlow is used to model VRF outdoor unit
        \note Supply Air Fan Object Type must be Fan:SystemModel or Fan:VariableVolume if
        \note AirConditioner:VariableRefrigerantFlow:FluidTemperatureControl or
        \note AirConditioner:VariableRefrigerantFlow:FluidTemperatureControl:HR
        \note is used to model VRF outdoor unit
        \note Required for zone equipment. Leave blank if terminal unit is used in AirLoopHVAC:OutdoorAirSystem:EquipmentList.
        \note Also leave blank if terminal unit is used on main AirloopHVAC branch and terminal unit has no fan.
```

I did verify that the fan in the curve-based VRF terminal unit can work properly in cycling-fan-cycling-coil. There might still be some issue in the `Fan:VariableVolume` as it doesn't involve the `state.dataHVACGlobal->OnOffFanPartLoadFraction` in its calculation (`SimVariableVolumeFan`). Since the curve-based VRF TU doesn't use the VariableVolume fan, the issue in `Fan:VariableVolume` is not triggered there. For the \*FluidTemperatureControl\* terminal units that use Fan:VariableVolume, there might still be some problem? @rraustad

Reply:
The VAV fan was never intended to cycle during the time step and so OnOffFanPartLoadFraction was never added to that fan model. When using a VAV fan, does the fan ever operate for only a portion of the time step?

Reply:
Yes. I believe the VAV fan will cycle. In the [issue description](https://github.com/NREL/EnergyPlus/issues/10106), the minimum fan power of 189.81W is the lowest the fan can operate without cycling. When the coil is cycling, the fan should also cycle and output less power.

Reply:
I'm a little concerned about adding OnOffFanPartLoadFraction to the VAV fan since it should run the entire time step. For other equipment the VAV fan can only be used with constant fan operating mode. You could certainly try it but this might cause more problems than it's worth. I am not sure if when you add OnOffFanPartLoadFraction to this fan model, if results for other configurations would change (e.g., VAV with a DX coil). If you change the fan model to OnOff do you get expected results? I am also concerned about the energy balance in the air stream which I don't think we have really focused on (i.e., HP cooling energy versus sum of TU cooling energy).

Reply:
I will try to remove the constraint of fan type and try the OnOff fan to make sure that it can properly cycle (Currently the `AirConditioner:VariableRefrigerantFlow:FluidTemperatureControl` won't allow OnOff fan). I'm guessing it could properly cycle but I will double check.

Another question is, is the following type of fix also considered bad? If so, and that adding OnOffFanPartLoadFraction to the VAV fan is also bad, do I create a separate fan type (similar to VAV but can cycle as well), and then use this type of fan for the variable volume fan in `AirConditioner:VariableRefrigerantFlow:FluidTemperatureControl`?
![image](https://github.com/NREL/EnergyPlus/assets/13421161/40a0ee11-6a63-4956-b92f-007b439ab21e)

Reply:
We don't need another fan type, just need to think this through carefully. I did think of using VAVOnOffFanPartLoadFraction only for the fluid cooler model but I don't like that either since we don't need more globals. An optional argument that defaults to 1 in the SimFan call may be another option, but we don't need more optional arguments. So keep thinking.

Reply:
![image](https://github.com/NREL/EnergyPlus/assets/13421161/33e76366-974b-485d-b008-94576771a763)
This is when the OnOff fan is used. The plot shows the fan power and cooling coil RTF when the system is in cooling mode.

Reply:
> We don't need another fan type, just need to think this through carefully. I did think of using VAVOnOffFanPartLoadFraction only for the fluid cooler model but I don't like that either since we don't need more globals. An optional argument that defaults to 1 in the SimFan call may be another option, but we don't need more optional arguments. So keep thinking.

Is optional arguments bad? Is overloading better (two versions, one with another argument)?

Reply:
That is an interesting fan power (flow) profile. I assume the fan is cycling below coil RTF = 0.2. I'd have to see fan power vs coil cycling ratio to be sure. I think you could pass coil PLF, based on coil cycling ratio, over to the fan and use that in the fan power calculation (using `* PLR/PLF`). I guess I would use the optional argument = 1 method for that but others could offer suggestions.

Reply:
I did some experiment using the non-fluid-control VRF model
[VariableRefrigerantFlow_5Zone.idf.txt](https://github.com/user-attachments/files/15796409/VariableRefrigerantFlow_5Zone.idf.txt)
[USA_NY_New.York-John.F.Kennedy.Intl.AP.744860_TMY3.epw.txt](https://github.com/user-attachments/files/15796414/USA_NY_New.York-John.F.Kennedy.Intl.AP.744860_TMY3.epw.txt)
I changed the "VRFFanSchedule" to always be 0.0 (fan cycling with the coil). I kinda expect the fan power to be linear with the coil runtime fraction. Do you feel this is expected?

![image](https://github.com/NREL/EnergyPlus/assets/13421161/3004ab61-ce8e-4751-b8bd-4c1534b3c6c1)

Reply:
I expect to see a curved arc above the linear line from 0,0 to 1,fan power. Something like this. Not sure what the group of data is at 0.9 RTF above 200 W. So this is closer to what I expect.

![image](https://github.com/NREL/EnergyPlus/assets/5778156/51fe6286-0a56-4d84-bbd8-3b09b6fdcb2e)

Reply:
This is the fan power-cooling coil RTF plot for VariableRefrigerantFlow_5Zone.idf during the whole year (restricted to cooling coil RTF not equal to 0 or 1). Again I used the Supply Air Fan Operating Mode Schedule is set to 0.

![image](https://github.com/NREL/EnergyPlus/assets/13421161/ac9dff92-0b6b-491c-a3fb-fd4dc3ad066c)

Reply:
> That is an interesting fan power (flow) profile. I assume the fan is cycling below coil RTF = 0.2. I'd have to see fan power vs coil cycling ratio to be sure. I think you could pass coil PLF, based on coil cycling ratio, over to the fan and use that in the fan power calculation (using `* PLR/PLF`). I guess I would use the optional argument = 1 method for that but others could offer suggestions.

I just pushed the commit using an operational argument to allow VAV fan to cycle in the VRFFuildTCtrl model. Please see if it looks okay.

Coil cycling ratio is not an existing output variable. I saw in the code that cooling coil runtime fraction is calculated as `thisDXCoil.CoolingCoilRuntimeFraction = CompCycRatio / PLF;` PLF is calculated with some curve 
```
        if (thisDXCoil.PLFFPLR(Mode) > 0 && CompCycRatio < 1.0) {
            PLF = CurveValue(state, thisDXCoil.PLFFPLR(Mode), CompCycRatio); // Calculate part-load factor
        } else {
            PLF = 1.0;
        }
```
However since object `Coil:Cooling:DX:VariableRefrigerantFlow:FluidTemperatureControl` doesn't have the `Part Load Fraction Correlation Curve Name` field, the PLF value is always 1. Cooling coil runtime fraction will just be the same as the compressor cycling ratio (and also coil cycling ratio)?

Reply:
The cycling of the fan should be the same as the coil/compressor. But looking at the figure, where fan power should be proportional to fan PLR (or fan air flow wrt max air flow), the fan PLR is much different from the coil PLR (coil PLR = coil RTF * coil PLF). Plot the fan power versus coil cycling ratio to see what that looks like. I'm still not sure what to pass to the fan model.

Reply:
The following plots the fan power and cooling coil cycling ratio. The fan her is a Fan:VariableVolume.
![image](https://github.com/NREL/EnergyPlus/assets/13421161/70a51228-f4ef-483d-8251-c48f19e7c577)
[eplusout_variableVolumeFan.xlsx](https://github.com/user-attachments/files/15876696/eplusout_variableVolumeFan.xlsx)

The following plots the relationship for the Fan:OnOff.
![image](https://github.com/NREL/EnergyPlus/assets/13421161/4d939fc8-9d13-4335-ad66-0769ee87cf6f)
[eplusout_onOffFan.xlsx](https://github.com/user-attachments/files/15876698/eplusout_onOffFan.xlsx)
@rraustad

Reply:
This is the Fan:OnOff results and must be using a unity PLF curve? Is this the fluid cooler model? I see coil RTF basically equal to HP cycling ratio which means no impact for startup losses. The only way I know that fan power can vary like this is if air flow is not proportional to PLR. I'm not saying that's bad, I just don't completely understand how air flow is controlled. If I assume air flow is independent of PLR, for example air flow is controlled to maintain a certain suction temperature, then I would think PLF should be passed to the fan and the fan calculate RTF = fan PLR/PLF.

![image](https://github.com/NREL/EnergyPlus/assets/5778156/729fa033-96cb-47af-b03e-fe6bf0b8fd2e)

Reply:
The coil object Coil:Cooling:DX:VariableRefrigerantFlow:FluidTemperatureControl doesn't have the Part Load Fraction Correlation Curve Name field, the PLF value is always 1. This is the Fluid cooling model indeed. I'll look at the airflow as well

Reply:
For the OnOff fan, the fan RTF and air mass flow both resemble the fan power shape.

The following is the fan air mass flow plotted against the cooling coil runtime fraction. 

![image](https://github.com/NREL/EnergyPlus/assets/13421161/062d77d3-5f95-4c9a-846a-899cab2d030b)

The following is the fan runtime fraction plotted against cooling coil runtime fraction
![image](https://github.com/NREL/EnergyPlus/assets/13421161/0792d664-14ca-459e-86e0-c57ee61a6e38)
[eplusout_onOffFan_fanAirFlow_vs_coilRTF.xlsx](https://github.com/user-attachments/files/15878200/eplusout_onOffFan_fanAirFlow_vs_coilRTF.xlsx)

[US+SF+CZ4A+hp+crawlspace+IECC_2006_VRF_onOffFan.idf.zip](https://github.com/user-attachments/files/15878204/US%2BSF%2BCZ4A%2Bhp%2Bcrawlspace%2BIECC_2006_VRF_onOffFan.idf.zip)
@rraustad

Reply:
> This is the Fan:OnOff results and must be using a unity PLF curve? Is this the fluid cooler model? I see coil RTF basically equal to HP cycling ratio which means no impact for startup losses. The only way I know that fan power can vary like this is if air flow is not proportional to coil PLR. I'm not saying that's bad, I just don't completely understand how air flow is controlled. If I assume air flow is independent of coil PLR, for example air flow is controlled to maintain a certain suction temperature, then I would think PLF should be passed to the fan and the fan calculate RTF = fan PLR/PLF.
> 
> ![image](https://private-user-images.githubusercontent.com/5778156/340450040-729fa033-96cb-47af-b03e-fe6bf0b8fd2e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk1MTA4MjQsIm5iZiI6MTcxOTUxMDUyNCwicGF0aCI6Ii81Nzc4MTU2LzM0MDQ1MDA0MC03MjlmYTAzMy05NmNiLTQ3YWYtYjAzZS1mZTZiZjBiOGZkMmUucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDYyNyUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA2MjdUMTc0ODQ0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9OTYzZGQ5NGZjYjMzNTZkYTc0NGQ3NzA4NjNiMGU1YzVjNGVlYzRiNTk1NmUzYjAwZWFhZmM3YzMzOTE5MmZjMiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.LbiVEcXiqtKiWr5pmixMNA9dTDXnZWOc4L6vTyvGLY0)

Since this PR is more about the variable volume fan (the only non-system-model one accepted by the FluidTCtrl model, maybe we can first try to figure this part out in the current PR and discuss whether to allow for OnOff fans in another separate PR? (or do you feel using OnOff fan instead of variable volume fan should be part of the fix?) Do you feel the current fix in [this commit](https://github.com/NREL/EnergyPlus/pull/10341/commits/9de83a18a2dc02f4b9a180a76a45fd3f5dda19bd) is okay or still not general enough? @rraustad

Reply:
This branch is fine for the fans allowed for this model. Fan:SystemModel or Fan:VariableVolume. I was just trying to figure out what the fan power and air flow should look like.

Reply:
I see